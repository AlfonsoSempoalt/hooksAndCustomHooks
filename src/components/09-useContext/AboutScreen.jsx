import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const AboutScreen = (props) => {
  const { user, setUser } = useContext(UserContext);
  const handleLogOut = () => {
    setUser({});
  };
  return (
    <>
      <h1>AboutScreen</h1>
      <hr />
      <pre>{JSON.stringify(user)}</pre>
      <button className='btn btn-warning' onClick={handleLogOut}>
        logout
      </button>
    </>
  );
};

export default AboutScreen;
