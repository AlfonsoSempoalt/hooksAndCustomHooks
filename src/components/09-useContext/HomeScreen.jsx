import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const HomeScreen = (props) => {
  const { user } = useContext(UserContext);
  console.log(user);
  return (
    <>
      <h1>Home page</h1>
      <hr />

      <pre>{ JSON.stringify( user )}</pre>
    </>
  );
};

export default HomeScreen;
