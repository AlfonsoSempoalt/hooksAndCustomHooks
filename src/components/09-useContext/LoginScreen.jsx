import React, { useContext } from 'react';
import { UserContext } from './UserContext';

const LoginScreen = (props) => {
  const { setUser } = useContext(UserContext);
  const handleLogIn = () => {
    setUser({
      id: 123,
      name: 'Alfonso Sempoalt',
    });
  };

  return (
    <>
      <h1>LoginScreen</h1>
      <hr />
      <button className='btn btn-primary mx-5' onClick={handleLogIn}>
        Log in
      </button>
    </>
  );
};

export default LoginScreen;
