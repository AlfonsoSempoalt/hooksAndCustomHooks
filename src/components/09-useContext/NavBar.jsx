import React from 'react';
import { Link, NavLink } from 'react-router-dom';

const NavBar = () => {
  return (
    <>
      <nav className='navbar navbar-expand-sm navbar-dark bg-dark'>
        <ul>
          <Link to='/' className='navbar-brand'>
            NavBar
          </Link>
          <section className='collapse navbar-collapse' id='navbarNavAltMarkup'>
            <div className='navbar-nav'>
              <li>
                <NavLink
                  activeClassName='active'
                  className='nav-link'
                  to='./'
                  exact
                >
                  Home
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName='active'
                  className='nav-link'
                  to='./about'
                  exact
                >
                  About
                </NavLink>
              </li>
              <li>
                <NavLink
                  activeClassName='active'
                  className='nav-link'
                  to='./login'
                  exact
                >
                  Log in
                </NavLink>
              </li>
            </div>
          </section>
        </ul>
      </nav>
    </>
  );
};

export default NavBar;
