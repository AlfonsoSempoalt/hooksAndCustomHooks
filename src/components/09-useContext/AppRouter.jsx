import React from 'react';
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
} from 'react-router-dom';
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LoginScreen from './LoginScreen';
import NavBar from './NavBar';

const AppRouter = () => {
  return (
    <>
      <Router>
        <section>
          <NavBar />

          <Switch>
            <Route exact={true} path='/about' component={AboutScreen} />
            <Route exact={true} path='/login' component={LoginScreen} />
            <Route exact={true} path='/' component={HomeScreen} />
            <Redirect to='/' />
          </Switch>
        </section>
      </Router>
    </>
  );
};

export default AppRouter;
