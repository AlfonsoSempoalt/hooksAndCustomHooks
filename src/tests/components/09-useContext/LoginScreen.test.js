import { mount } from 'enzyme';
import React from 'react';
import LoginScreen from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('tests to <LoginScreen>', () => {
    const setUser = jest.fn();

    let wrapper = mount(
        <UserContext.Provider
            value={{
                setUser,
            }}
        >
            <LoginScreen />
        </UserContext.Provider>
    );

    beforeEach(() => {
        wrapper = mount(
            <UserContext.Provider
                value={{
                    setUser,
                }}
            >
                <LoginScreen />
            </UserContext.Provider>
        );
    });

    test('should match with the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should logIn properly', () => {
        const user = {
            id: 123,
            name: 'Alfonso Sempoalt',
        };
        wrapper.find('button').simulate('click');
        expect(setUser).toHaveBeenCalledWith(user);
    });
});
