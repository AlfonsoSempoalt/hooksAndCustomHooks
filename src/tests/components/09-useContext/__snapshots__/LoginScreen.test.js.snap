// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests to <LoginScreen> should match with the snapshot 1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <button
    className="btn btn-primary mx-5"
    onClick={[Function]}
  >
    Log in
  </button>,
]
`;
